examples/gno.land/p/demo/diff/diff.gno

```diff
@@ -0,0 +1,151 @@
+package diff
+
+import (
+	"strings"
+)
+
+type EditType int
+
+const (
+	EditKeep EditType = iota
+	EditInsert
+	EditDelete
+)
+
+type Edit struct {
+	Type EditType
+	Char rune
```

Comment:
```suggestion
	Char byte
```

Why not using bytes instead of runes, which make the diff algorithm applicable to any kind of data (including binary), not just strings. It should not perclude the correct display of string data prior/after the diff.

Storage size: the `Edit` struct requires 128 bits per item (64 bits for `EditType`, 32 bits for `Char` aligned to 64 bits). after having `EditType` as an `int8`  or `uint8`, and `Char` as a `byte`, `Edit` struct just requires 16 bits, 8 times more efficient for the same code :-)

Reply:
Thank you or your detasiled review. First, the use of `rune` instead of `byte` type was to handle non-ASCII characters (especially Unicode). For example, in Korean, the character '가' is encoded as 3 bytes in UTF-8, and if processed as a single byte, the character would be corrupted. However, this problem can be resolved by using `rune`, which is why this type has been used. It's a kind of i18n.

Anyway, it seems that applying the other suggestions wouldn't cause any problems. So, I think it would be good to keep only the `rune` type as is, and modify the rest. What do you think?

Reply:
> It would be nice to have an additional test involving multi-byte chars to demonstrate that. Thanks

Added some tests. For grapheme clusters composed of emoji combinations or case of Unicode normalization, there are occasional broken in some cases. But it work well in most common cases.

I also added cases to check for worst-case scenarios, generating test strings by assembling them using `strings.Repeat` function. However, when coparing the speed with Go, there is a significant difference. I think need to investigate separately later. Luckily, this does not prone any problems for tis testing testing.

https://github.com/gnolang/gno/pull/2490/commits/71d1affa24e2356cababe6a596c28ac71dfbe9a4